### PR TITLE
dbus: upstream Linux fixes

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
These failures all appear to be related to issues with `qt`, not this PR.  If we think the changes look good, we should get this merged because other Linux-only formula depend on it.

**Comment by carlocab** [comment]
@paperchalice is there a `qt` flag to prevent linkage with `brotli`?

**Comment by carlocab** [comment]
#75926

**Comment by iMichka** [comment]
Wondering if we should disable the docs completely? We often do not ship docs. What was the error on Linux?

**Comment by paperchalice** [comment]
> @paperchalice is there a `qt` flag to prevent linkage with `brotli`?

Try `-no-feature-brotil`, module `network` may need it, `qtwebengine` in future may also need it.

**Comment by carlocab** [comment]
@iMichka docs added in #6835. Seems like a user explicitly asked for them, and the docs are apparently not easily available online? This was a while ago, though, so things might be different now. If we can verify that the situation re doc availability elsewhere has changed, we can remove the docs.

@paperchalice already added the flag in #75926. Or should we just add the `brotli` dep, if it might actually be needed?

**Comment by danielnachun** [comment]
Everything in the `share/doc` of the `dbus` cellar on macOS can be found here: https://dbus.freedesktop.org/doc/, along with much more.  Also, there are no `.xml` files even present in the cellar, so `--enable-xml-docs` doesn't even seem to produce anything.  If there are no objections, I'm just going to use `--disable-xml-docs` for both platforms.

**Comment by danielnachun** [comment]
Scratch what I said above - `--enable-xml-docs` is needed to generate the man pages (I just tested this locally), which I think is something we like to provide to users if possible.  IMO that wasn't a great choice of argument name by upstream.  

I'm actually thinking instead that we may want to make `xmlto` a dependency on Linux as well and use `--enable-xml-docs` on both platforms.  I'll have to test that on Linux to make sure it works.

**Comment by paperchalice** [comment]
Adding `depends_on "brotli"` seems not bad, since `brotli` is a small package.

**Comment by carlocab** [comment]
Merging #75926. You can rebase this when that's done.

File: Formula/dbus.rb
Comment (carlocab) [inline Formula/dbus.rb:56]: We shouldn't need this if you make `pkg-config` a build dependency on macOS too. I think taking out two `on_os` blocks makes the added dependency worth it.

Comment (danielnachun) [reply Formula/dbus.rb]: I tested this locally and adding `pkg-config` as a build dependency for both platforms did allow me to remove those lines.  This really cleans up the formula nicely!

**Comment by danielnachun** [comment]
I added the missing license info and everything now passes in CI.  Can I get a final review before merging this?

**Review by carlocab — APPROVED** [review]

**Pull request merged**
